### PR TITLE
Updated support for Cisco NXOS devices

### PR DESCRIPTION
**Pull request by ReK42**
##### ISSUE TYPE
 - New Template Pull Request

##### COMPONENT
- `show_ip_ospf_interface_brief`
- `show_ip_pim_rp_vrf`
- `show_ip_pim_neighbor_vrf`
- `show_ip_pim_interface_brief_vrf`
- `show_ip_pim_group-range_vrf`

##### SUMMARY
Added 5 new commands.

See tests for command output.

### CODE REVIEW

**Comment by ReK42** [comment]
There appears to be a problem with the Python 3.7 test environment, Travis is showing `The command "poetry install" failed`. Tests are passing for 3.6 and 3.8.

**Comment by jvanderaa** [comment]
Restarted the build that failed still. 

Will continue to check on that, but also missing the `ERROR` statement.

**Comment by ReK42** [comment]
Looks like the environment is failing to be installed:

```
• Installing future (0.18.2): Installing...

  • Installing mccabe (0.6.1): Failed

  EnvCommandError

  Command ['/home/travis/virtualenv/python3.7.1/bin/pip', 'install', '--no-deps', '/home/travis/.cache/pypoetry/artifacts/c2/cd/ed/3c4495a1422fb12eefbca8b3c6ccc83ab4ec92fd39df029199cc4f4ee4/mccabe-0.6.1-py2.py3-none-any.whl'] errored with the following return code 1, and output: 

  Processing /home/travis/.cache/pypoetry/artifacts/c2/cd/ed/3c4495a1422fb12eefbca8b3c6ccc83ab4ec92fd39df029199cc4f4ee4/mccabe-0.6.1-py2.py3-none-any.whl

  Installing collected packages: mccabe

  ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/packaging-20.9.dist-info'
```

**Review by pszulczewski — CHANGES_REQUESTED** [review]
@ReK42 can you re-base this PR on master.
CI is now fixed.

**Comment by ReK42** [comment]
It looks like someone else added one of the commands in the meantime. I've removed that but kept the `show ip pim` commands. Note that due to the length of time since the original PR I no longer have an environment to test this in.

File: ntc_templates/templates/cisco_nxos_show_ip_pim_group-range_vrf_all.textfsm [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Value MODE (\S+)
+Value RP_ADDRESS (\S+)
+Value SHARED_TREE_RANGE (\S+)
+Value ORIGIN (\S+)
+
+
+Start
+  ^\s*PIM Group-Range Configuration for VRF "${VRF}"\s*$$
+  ^\s*Group-range\s+Action\s+Mode\s+RP-address\s+Shrd-tree-range\s+Origin\s*$$
+  ^\s*${GROUP_RANGE}\s+${ACTION}\s+${MODE}\s+${RP_ADDRESS}\s+${SHARED_TREE_RANGE}\s+${ORIGIN}\s*$$ -> Record
```
Comment (pszulczewski) [inline ntc_templates/templates/cisco_nxos_show_ip_pim_group-range_vrf_all.textfsm:13]:
```suggestion
  ^\s*${GROUP_RANGE}\s+${ACTION}\s+${MODE}\s+${RP_ADDRESS}\s+${SHARED_TREE_RANGE}\s+${ORIGIN}\s*$$ -> Record
  ^. -> Error
 
```

File: ntc_templates/templates/cisco_nxos_show_ip_pim_interface_brief_vrf_all.textfsm [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Value IP_ADDRESS (\d+\.\d+\.\d+\.\d+)
+Value PIM_DR_ADDRESS (\d+\.\d+\.\d+\.\d+)
+Value NEIGHBOR_COUNT (\d+)
+Value BORDER_INTERFACE (\S+)
+
+Start
+  ^\s*PIM Interface Status for VRF "${VRF}"\s*$$
+  ^\s*Interface\s+IP Address\s+PIM DR Address\s+Neighbor\s+Border\s*$$
+  ^\s*Count\s+Interface\s*$$
+  ^\s*${INTERFACE}\s+${IP_ADDRESS}\s+${PIM_DR_ADDRESS}\s+${NEIGHBOR_COUNT}\s+${BORDER_INTERFACE}\s*$$ -> Record
```
Comment (pszulczewski) [inline ntc_templates/templates/cisco_nxos_show_ip_pim_interface_brief_vrf_all.textfsm:12]:
```suggestion
  ^\s*${INTERFACE}\s+${IP_ADDRESS}\s+${PIM_DR_ADDRESS}\s+${NEIGHBOR_COUNT}\s+${BORDER_INTERFACE}\s*$$ -> Record
  ^. -> Error

```

File: ntc_templates/templates/cisco_nxos_show_ip_pim_neighbor_vrf_all.textfsm [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+Value BIDIR_CAPABLE (\S+)
+Value BFD_STATE (\S+)
+Value ECMP_REDIRECT_CAPABLE (\S+)
+
+
+Start
+  ^\s*PIM Neighbor Status for VRF "${VRF}"\s*$$
+  ^\s*Neighbor\s+Interface\s+Uptime\s+Expires\s+DR\s+Bidir-\s+BFD\s+ECMP Redirect\s*$$
+  ^\s*Priority\s+Capable\s+State\s+Capable\s*$$
+  ^\s*${NEIGHBOR}\s+${INTERFACE}\s+${UPTIME}\s+${EXPIRES}\s+${DR_PRIORITY}\s+${BIDIR_CAPABLE}\s+${BFD_STATE}\s+${ECMP_REDIRECT_CAPABLE}\s*$$ -> Record
```
Comment (pszulczewski) [inline ntc_templates/templates/cisco_nxos_show_ip_pim_neighbor_vrf_all.textfsm:16]:
```suggestion
  ^\s*${NEIGHBOR}\s+${INTERFACE}\s+${UPTIME}\s+${EXPIRES}\s+${DR_PRIORITY}\s+${BIDIR_CAPABLE}\s+${BFD_STATE}\s+${ECMP_REDIRECT_CAPABLE}\s*$$ -> Record
  ^. -> Error

```

File: ntc_templates/templates/cisco_nxos_show_ip_pim_rp_vrf_all.textfsm [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+Value AUTO_RP_DISCOVERY_POLICY (\S+)
+
+Start
+  ^\s*PIM RP Status Information for VRF "${VRF}"\s*$$
+  ^\s*BSR ${BSR}\s*$$
+  ^\s*Auto-RP ${AUTO_RP}\s*$$
+  ^\s*BSR RP Candidate policy: ${BSR_CANDIDATE_POLICY}\s*$$
+  ^\s*BSR RP policy: ${BSR_POLICY}\s*$$
+  ^\s*Auto-RP Announce policy: ${AUTO_RP_ANNOUNCE_POLICY}\s*$$
+  ^\s*Auto-RP Discovery policy: ${AUTO_RP_DISCOVERY_POLICY}\s*$$ -> Record
```
Comment (pszulczewski) [inline ntc_templates/templates/cisco_nxos_show_ip_pim_rp_vrf_all.textfsm:16]:
```suggestion
  ^\s*Auto-RP Discovery policy: ${AUTO_RP_DISCOVERY_POLICY}\s*$$ -> Record
  ^. -> Error

```

**Review by pszulczewski — CHANGES_REQUESTED** [review]
Please add `  ^. -> Error` at the end of each template so there is an error raised instead of an empty result and we are good to merge ;-)
Thank you!

File: ntc_templates/templates/cisco_nxos_show_ip_pim_group-range_vrf_all.textfsm [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  ^. -> Error
+
```
Comment (pszulczewski) [inline ntc_templates/templates/cisco_nxos_show_ip_pim_group-range_vrf_all.textfsm:15]:
```suggestion
  ^. -> Error

```

**Review by pszulczewski — CHANGES_REQUESTED** [review]

**Review by pszulczewski — APPROVED** [review]
OK looks like I was able to make the necessary changes myself.

Many thanks for your contribution and apologies that it took so long to merge it.

**Pull request merged**